lib/mini_racer.rb

```diff
@@ -143,16 +143,18 @@ def initialize(name, callback, parent)
       end
     end
 
-    def initialize(max_memory: nil, timeout: nil, isolate: nil, ensure_gc_after_idle: nil, snapshot: nil, marshal_stack_depth: nil)
+    def initialize(max_memory: nil, timeout: nil, cputime_limit: nil, isolate: nil, ensure_gc_after_idle: nil, snapshot: nil, marshal_stack_depth: nil)
       options ||= {}
 
-      check_init_options!(isolate: isolate, snapshot: snapshot, max_memory: max_memory, marshal_stack_depth: marshal_stack_depth, ensure_gc_after_idle: ensure_gc_after_idle, timeout: timeout)
+      check_init_options!(isolate: isolate, snapshot: snapshot, max_memory: max_memory, marshal_stack_depth: marshal_stack_depth, ensure_gc_after_idle: ensure_gc_after_idle, timeout: timeout, cputime_limit: cputime_limit)
 
       @functions = {}
       @timeout = nil
+      @cputime_limit = nil
       @max_memory = nil
       @current_exception = nil
       @timeout = timeout
+      @cputime_limit = cputime_limit
       @max_memory = max_memory
```

Comment:
Why are these ivar inits duplicated? That shouldn't be necessary, I think... (and we could skip it on our own limit; much  like `@marshal_stack_depth` does)

Reply:
Yeah, I duped it to match the other stuff. I was also confused, but if there's a reason -- =operator side effect that we depend on or something, I didn't want to introduce that issue. 😬 

But if there's an example of one not doing it, I guess we can, too. (unfortunately that example is also code we added, so...) A good test though is to check latest main mini_racer-- they may have already fixed this multi-ivar init.

Reply:
Ruby doesn't have a `=operator` to overload (though `a.foo = b` desugars to `a.foo=(b)`; but that's not happening here, this is a variable assignment, `a = b`; [cf. ruby docs](https://ruby-doc.org/docs/ruby-doc-bundle/Manual/man-1.4/syntax.html#assign)).

Mainline MiniRacer still has the double-init; though looking at a blame this code hasn't been touched in 3 years (4 years if we exclude your contribution). It *looks* like it's written this way because originally some assignments (e.g. `@max_memory`) were conditional (type checks on an `options` Hash member); [cf. diff](https://github.com/rubyjs/mini_racer/commit/4f40f41ce48716541b1ea85ad765dc8908d7da12#diff-c5d68985c8ecdf7f793874226c4bd2df2991cba0bba6a395ffbb77813eb4db09L143).

I'd personally say we don't need to replicate it with our variables, but I can see a consistency argument.

Reply:
> Ruby doesn't have a `=operator` to overload (though `a.foo = b` desugars to `a.foo=(b)`; but that's not happening here, this is a variable assignment, `a = b`; [cf. ruby docs](https://ruby-doc.org/docs/ruby-doc-bundle/Manual/man-1.4/syntax.html#assign)).
> 
> Mainline MiniRacer still has the double-init; though looking at a blame this code hasn't been touched in 3 years (4 years if we exclude your contribution). It _looks_ like it's written this way because originally some assignments (e.g. `@max_memory`) were conditional (type checks on an `options` Hash member); [cf. diff](https://github.com/rubyjs/mini_racer/commit/4f40f41ce48716541b1ea85ad765dc8908d7da12#diff-c5d68985c8ecdf7f793874226c4bd2df2991cba0bba6a395ffbb77813eb4db09L143).
> 
> I'd personally say we don't need to replicate it with our variables, but I can see a consistency argument.

Let's leave out the = nil first assignments for now and address it if necessary when we move our main branch forward (and update ruby etc etc)

Reply:
removed.